x-pack/plugins/secops/public/components/page/network/ip_overview/field_renderers.test.tsx

```diff
@@ -0,0 +1,226 @@
+ */
+
+import { mount } from 'enzyme';
+import toJson from 'enzyme-to-json';
+import * as React from 'react';
+import AutonomousSystem = GetIpOverviewQuery.AutonomousSystem;
+import { mountWithIntl } from 'test_utils/enzyme_helpers';
+
+import { GetIpOverviewQuery, HostEcsFields, IpOverviewType } from '../../../../graphql/types';
+import { TestProviders } from '../../../../mock';
+
+import {
+  autonomousSystemRenderer,
+  dateRenderer,
+  hostNameRenderer,
+  locationRenderer,
+  whoisRenderer,
+} from './field_renderers';
+import { mockData } from './mock';
+
+describe('Field Renderers', () => {
+  describe('#locationRenderer', () => {
+    test('it renders correctly against snapshot', () => {
+      const wrapper = mount(
+        <TestProviders>
+          {locationRenderer(['source.geo.city_name', 'source.geo.region_name'], mockData.complete)}
+        </TestProviders>
+      );
+
+      expect(toJson(wrapper)).toMatchSnapshot();
+    });
+
+    test('it renders emptyTagValue when no fields provided', () => {
+      const wrapper = mount(
+        <TestProviders>{locationRenderer([], mockData.complete)}</TestProviders>
+      );
+      expect(wrapper.text()).toEqual('--');
+    });
+
+    test('it renders emptyTagValue when invalid fields provided', () => {
+      const wrapper = mount(
+        <TestProviders>
+          {locationRenderer(['source.geo.my_house'], mockData.complete)}
+        </TestProviders>
+      );
+      expect(wrapper.text()).toEqual('--');
+    });
+  });
+
+  describe('#dateRenderer', () => {
+    test('it renders correctly against snapshot', () => {
+      const wrapper = mount(
+        <TestProviders>{dateRenderer('firstSeen', mockData.complete.source!)}</TestProviders>
+      );
+
+      expect(toJson(wrapper)).toMatchSnapshot();
+    });
+
+    test('it renders emptyTagValue when non-string fieldprovided', () => {
+      const wrapper = mount(
+        <TestProviders>{dateRenderer('geo.location.lat', mockData.complete.source!)}</TestProviders>
+      );
+      expect(wrapper.text()).toEqual('--');
+    });
+
+    test('it renders emptyTagValue when invalid field provided', () => {
+      const wrapper = mount(
+        <TestProviders>{dateRenderer('geo.spark_plug', mockData.complete.source!)}</TestProviders>
+      );
+      expect(wrapper.text()).toEqual('--');
+    });
+  });
+
+  describe('#autonomousSystemRenderer', () => {
+    const emptyMock: AutonomousSystem = { as_org: null, asn: null, ip: '10.10.10.10' };
+    const halfEmptyMock: AutonomousSystem = { as_org: null, asn: 'Test ASN', ip: '10.10.10.10' };
+
+    test('it renders correctly against snapshot', () => {
+      const wrapper = mount(
+        <TestProviders>
+          {autonomousSystemRenderer(
+            mockData.complete.source!.autonomousSystem!,
+            IpOverviewType.source
+          )}
+        </TestProviders>
+      );
+
+      expect(toJson(wrapper)).toMatchSnapshot();
```

Comment:
I'd use `shallow` for snapshot or if that doesn't make sense remove it as mount makes deep compares and very large snapshots.